src/metabase/models/recent_views.clj

```diff
@@ -69,16 +72,30 @@
                 ;;       distinguish between them.
    :dashboard :table :collection])
 
-(defn- ids-to-prune-for-user+model [user-id model]
+(defn- moi->type
+  "Turns a model of interest into the base name of its table"
+  [moi]
+  (if (#{:model :card :metric} moi)
+    "card"
+    (name moi)))
+
+(mu/defn moi->model
+  "Given a model of interest, returns the model for it."
+  [moi :- (into [:enum] models-of-interest)]
+  (keyword "model" (str/capitalize (moi->type moi))))
+
+(defn- ids-to-prune-for-user+model [user-id model context]
   (t2/select-fn-set :id
                     :model/RecentViews
                     {:select [:rv.id]
                      :from [[:recent_views :rv]]
                      :where [:and
                              [:= :rv.model (get {:model "card"} model (name model))]
                              [:= :rv.user_id user-id]
-                             (when (#{:card :model} model)
+                             [:= :rv.context (h2x/literal (name context))]
+                             (when (#{:card :model} model) ;; TODO add metric
```

Comment:
Some driveby work has been done to include metrics in our work, but I don't think it ever worked, and we do not query for them, so this should be addressed at some point in the future.

Reply:
Could you add a little more context in a comment or the ns docstring or something? I'm imagining viewing this ns in a year and not being able to figure out why these TODOs are here

Reply:
Good call, done.